### PR TITLE
*：fix the bug about auto-increment key after renaming a table from old DB to the another DB

### LINKED ISSUE #4985: Rename a table with auto-increment key from old DB to the other DB, then drop the old DB
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
```
tidb> create table t(a int primary key auto_increment);
Query OK, 0 rows affected (0.01 sec)
tidb> insert t values();
Query OK, 1 row affected (0.01 sec)
tidb> select * from t;
+---+
| a |
+---+
| 1 |
+---+
1 row in set (0.00 sec)
tidb> create database test;
Query OK, 0 rows affected (0.00 sec)
tidb> rename table t to test.t;
Query OK, 0 rows affected (0.01 sec)
tidb> use test;
Database changed
tidb> drop database db;
Query OK, 0 rows affected (0.01 sec)
tidb> insert t values();
```

2. What did you expect to see?
```
Query OK, 1 row affected (0.01 sec)
```

3. What did you see instead?
```
ERROR 1062 (23000): Duplicate entry '1' for key 'PRIMARY'
```

### PR DESCRIPTION
**Pull request by zimulala**
Rebase auto ID in `table.go` when doing the operation of `rename table`. We shouldn't consider auto ID key in other places. It's more easy and readable. 
Now we don't remove the field of `OldSchemaID` because we need to be compatible with the old version.
Fix #4985

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: ddl/table.go [hunk 1]
```diff
@@ -233,18 +233,25 @@ func (d *ddl) onRenameTable(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 	if err != nil {
 		return ver, errors.Trace(err)
 	}
+	var baseID int64
 	newSchemaID := job.SchemaID
 	if newSchemaID != oldSchemaID {
 		err = checkTableNotExists(t, job, newSchemaID, tblInfo.Name.L)
 		if err != nil {
+			job.State = model.JobStateCancelled
```
Comment (shenli) [inline ddl/table.go]: This is a bug fix?

File: meta/meta.go [hunk 2]
```diff
@@ -292,32 +291,24 @@ func (m *Meta) DropDatabase(dbID int64) error {
...
+	tableKey := m.tableKey(tblID)
 	if err := m.checkTableExists(dbKey, tableKey); err != nil {
 		return errors.Trace(err)
 	}
 
 	if err := m.txn.HDel(dbKey, tableKey); err != nil {
 		return errors.Trace(err)
 	}
-
-	if delAutoID {
```
Comment (shenli) [inline meta/meta.go]: Should we keep the auto-id key for renaming table operation？

Comment (zimulala) [reply meta/meta.go]: No.
We will use the new auto-id key after renaming table. So we can delete it.

**Comment by zimulala** [comment]
PTAL @shenli

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
Related with https://github.com/pingcap/tidb/pull/3493/files

**Pull request merged**